### PR TITLE
[FLINK-6164] Make ProcessWindowFunction a RichFunction

### PR DESCRIPTION
**Pull request by zentol**
This PR modifies the ProcessWindowFunction and ProcessAllWindowFunction classes to extend AbstractRichtFunction.

The existing Rich* versions were deprecated, and all usages of them replaced.

### CODE REVIEW

File: flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/function/ProcessAllWindowFunction.scala [hunk 1]
```diff
@@ -35,7 +35,9 @@ import org.apache.flink.util.Collector
   * @tparam W The type of the window.
   */
 @PublicEvolving
-abstract class ProcessAllWindowFunction[IN, OUT, W <: Window] extends Function with Serializable {
+abstract class ProcessAllWindowFunction[IN, OUT, W <: Window]
```
Comment (aljoscha) [inline flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/function/ProcessAllWindowFunction.scala]: I think the `Serializable` is not needed here. (Was there before, so not introduced in this PR)

File: flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/function/ProcessWindowFunction.scala [hunk 2]
```diff
@@ -36,7 +36,10 @@ import org.apache.flink.util.Collector
   * @tparam W The type of the window.
   */
 @PublicEvolving
-abstract class ProcessWindowFunction[IN, OUT, KEY, W <: Window] extends Function with Serializable {
+abstract class ProcessWindowFunction[IN, OUT, KEY, W <: Window]
```
Comment (aljoscha) [inline flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/function/ProcessWindowFunction.scala]: See above.

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/windowing/AccumulatingAlignedProcessingTimeWindowOperatorTest.java [hunk 3]
```diff
@@ -1038,7 +1038,7 @@ private void assertInvalidParameter(long windowSize, long windowSlide) {
 
 	// ------------------------------------------------------------------------
 
-	private static class StatefulFunction extends RichProcessWindowFunction<Integer, Integer, Integer, TimeWindow> {
+	private static class StatefulFunction extends ProcessWindowFunction<Integer, Integer, Integer, TimeWindow> {
```
Comment (aljoscha) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/windowing/AccumulatingAlignedProcessingTimeWindowOperatorTest.java]: With this we add two unused imports.

And yes, I don't like that we don't have checkstyle active on the tests.

**Comment by zentol** [comment]
I've removed Serializable, adjusted the `@Deprecated` annotations like in #3816 and fixed the unused imports (there were also some in `ProcessAllWindowFunction.scala`).

**Comment by aljoscha** [comment]
Ha! I didn't think of the annotations.

This looks good to merge now!

Thanks for fixing these so quickly! 😃

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**